nixos/modules/services/continuous-integration/hercules-ci-agent/common.nix

```diff
@@ -0,0 +1,211 @@
+/*
+
+This file is for options that NixOS and nix-darwin have in common.
+
+Platform-specific code is in the respective default.nix files.
+
+ */
+
+{ config, lib, options, pkgs, ... }:
+
+let
+  inherit (lib) mkOption mkIf types filterAttrs literalExample;
+
+  cfg =
+    config.services.hercules-ci-agent;
+
+  format = pkgs.formats.toml {};
+
+  checkNix =
```

Comment:
I'm not a fan of something failing at build time here.

What about checking the nix version and failing if it's too low at evaluation time, and prompting users to use `nixUnstable` in that case?

Reply:
I would like to avoid depending on unreleased versions, which is why I've created the option to apply a minimal patch on the released version.
Deriving this information from options only seems a bit risky. For example if the user has their own patched Nix, how do we check that it's ok?

It's only a temporary solution. Once 2.4 is released, I'm happy to reduce this to a simple eval-only check.

Reply:
Hmmh, this code also depends on unreleased versions. Contrary to `nixUnstable`, you build your own version on-the-fly here, which is even less risky for sudden build failure.

I'd assume Nix 2.4 will be released before 19.10 - mind just recommending `nixUnstable` in an evaluation failure/warning (depending on how required it is) here if `!(lib.versionAtLeast (lib.getVersion config.nix.package))`?

Reply:
The patch has been tested since 2.3.4 and has not needed any changes for the total of four releases. It's a patch on the stable branch after all. 
I'd be happy to remove the patching when 2.4 is released. But I wouldn't just assume it's released before NixOS.